README.md

```diff
@@ -1,65 +1,73 @@
-
-## Pieces involved
-The central pieces of this solution are [Jenkins](https://jenkins.io/), [the Jenkins Job-DSL plugin](https://wiki.jenkins-ci.org/display/JENKINS/Job+DSL+Plugin), [Gogs](https://gogs.io/) and [Docker](https://www.docker.com/).  
+## Motivation
+Having a way of rapidly creating and deploying microservices is a prerequisite if you want to do serious microservices. It should be easy, so that we can focus on business, the thing that makes us unique.  
+This project serves as a starting point for automating all the operational setup cost associated to creating new microservices with a single click.
 
-We use Docker to run Jenkins and Gogs, and while running Jenkins we provide a way for it to run Docker itself (first inception...) with a Docker-outside-of-Docker approach (we map the host's Docker socket, so that containers started by Jenkins will be siblings).  
+## How it works
+The central pieces of this solution are [Docker](https://www.docker.com/), [Jenkins](https://jenkins.io/), [the Jenkins Job-DSL plugin](https://wiki.jenkins-ci.org/display/JENKINS/Job+DSL+Plugin) and [Gogs](https://gogs.io/) (it should be easy for you to adapt it to work with any other Git Server though, like [GitHub](https://github.com/) or [BitBucket](https://bitbucket.org)). 
 
-Within Gogs, we create two git repositories: the microservice_code_generator, which is self-explanatory, and the jenkins_seed, where the Jenkins Job-DSL definitions will live. The Jenkins Job-DSL plugin allows for having pipelines as code, in case you were not familiar with it.  
+We use Docker to run Jenkins and Gogs. Jenkins itself has access to Docker in a Docker-outside-of-Docker approach, i.e. we map the host's Docker socket, so that containers started by Jenkins will be siblings.
 
-Now, (our second inception...) using a seed job in Jenkins with the jenkins_seed repository as remote url (origin), we create the other jobs involved in this solution. Nice! The seed job's duty does not end here though, as it's responsible for keeping Jenkins pipelines updated as things evolve.  
+Being Gogs our Git Server, we push two repositories into it:  
+- microservice_code_generator, which is responsible for the generation of the microservices' skeleton.  
+We are creating simple Python Flask microservices in this project but it should be fairly simple for you to adapt it to your needs.
+- jenkins_seed, where we have our pipelines as code via Jenkins Job-DSL definitions.
 
-From that point, we already have a create_microservice pipeline that fulfills the promise and allows us to create microservices with.one.click! Really cool. How it does it, well, you can bet there's more inception coming... and you'd win. :)  
+One of those pipelines that we have is the create_microservice pipeline that fulfills the promise and creates microservices with.one.click. Awesome. What it does:
```

Comment:
Suggestion:
"One of the pipelines is the create_microservice pipeline that fulfills the promise of creating microservices with one click. What it does:"